OpenRA.Mods.Common/Activities/FindAndDeliverResources.cs

```diff
@@ -128,13 +113,13 @@ public override bool Tick(Actor self)
 			// of the refinery entrance.
 			if (LastSearchFailed)
 			{
-				var lastproc = harv.LastLinkedProc ?? harv.LinkedProc;
-				if (lastproc != null && !lastproc.Disposed)
+				var lastproc = harv.DockClientManager.LastReservedHost;
+				if (lastproc != null)
 				{
-					var deliveryLoc = lastproc.Location + lastproc.Trait<IAcceptResources>().DeliveryOffset;
-					if (self.Location == deliveryLoc && harv.IsEmpty)
+					var deliveryLoc = lastproc.DockPosition;
+					if (self.CenterPosition == deliveryLoc && harv.IsEmpty)
```

Comment:
Could you make this use a `DockingPosition` var that includes an optional offset instead of using the unit's `CenterPosition` directly? I am thinking of a dock where different harvester units will want different offsets so that they fit properly inside the refinery. E.g. a very large harvester might want a -20 X and Y offset to their DockingPosition, while a smaller harvester won't need any offset.